python/cudf/cudf/tests/utils.py

```diff
@@ -113,6 +115,82 @@ def assert_neq(left, right, **kwargs):
         raise AssertionError
 
 
+def assert_exceptions_equal(
+    lfunc,
+    rfunc,
+    lfunc_args_and_kwargs=None,
+    rfunc_args_and_kwargs=None,
+    compare_error_message=True,
+):
+    """Compares if two functions ``lfunc`` and ``rfunc`` raise
+    same exception or not.
+
+    Parameters
+    ----------
+    lfunc : callable
+        A callable function to obtain the Exception.
+    rfunc : callable
+        A callable function to compare the Exception
+        obtained by calling ``rfunc``.
+    lfunc_args_and_kwargs : tuple, default None
+        Tuple containing positional arguments at first position,
+        and key-word arguments at second position that need to be passed into
+        ``lfunc``.
+    rfunc_args_and_kwargs : tuple, default None
+        Tuple containing positional arguments at first position,
+        and key-word arguments at second position that need to be passed into
+        ``rfunc``.
+    compare_error_message : boolean, default True
+        Whether to compare the error messages raised
+        when calling both ``lfunc`` and
+        ``rfunc`` or not.
+
+    Returns
+    -------
+    None
+        If exceptions raised by ``lfunc`` and
+        ``rfunc`` match.
+
+    Raises
+    ------
+    AssertionError
+        If call to ``lfunc`` doesn't raise any Exception.
+    """
+
+    lfunc_args, lfunc_kwargs = _get_args_kwars_for_assert_exceptions(
+        lfunc_args_and_kwargs
+    )
+    rfunc_args, rfunc_kwargs = _get_args_kwars_for_assert_exceptions(
+        rfunc_args_and_kwargs
+    )
+
+    try:
+        lfunc(*lfunc_args, **lfunc_kwargs)
+    except KeyboardInterrupt:
+        raise
+    except Exception as e:
+        with pytest.raises(
+            type(e),
+            match=re.escape(str(e)) if compare_error_message is True else None,
+        ):
+            rfunc(*rfunc_args, **rfunc_kwargs)
+    else:
+        raise AssertionError("Expected to fail with an Exception.")
+
+
+def _get_args_kwars_for_assert_exceptions(func_args_and_kwargs):
+    if func_args_and_kwargs is None:
+        return [], {}
+    else:
+        if len(func_args_and_kwargs) == 1:
+            func_args, func_kwargs = func_args_and_kwargs[0], {}
```

Comment:
Can we tweak this a bit to make it easier when we only have kwargs? Will need to import `Mapping` and `Sequence` from `collections.abc`:
```suggestion
        if len(func_args_and_kwargs) == 1:
            func_args, func_kwargs = [], {}
            if isinstance(func_args_and_kwargs[0], Sequence):
                func_args = func_args_and_kwargs[0]
            elif isinstance(func_args_and_kwargs[0], Mapping):
                func_kwargs = func_args_and_kwargs[0]
            else:
                raise ValueError("length 1 func_args_and_kwargs must be either a Sequence or a Mapping")
```

Reply:
Updated the docs too so we clearly call out we support single length tuple